iree/samples/simple_embedding/simple_embedding.c

```diff
@@ -16,18 +16,24 @@
 #include "iree/vm/api.h"
 #include "iree/vm/bytecode_module.h"
 
-// Compiled module embedded here to avoid file IO:
-#if IREE_ARCH_RISCV_64
-#include "iree/samples/simple_embedding/simple_embedding_test_llvm_aot_rv64.h"
-#else
-#include "iree/samples/simple_embedding/simple_embedding_test_bytecode_module_c.h"
-#endif
-
 // A function to create the HAL device from the different backend targets.
 // The HAL device is returned based on the implementation, and it must be
 // released by the caller.
 extern iree_status_t create_sample_device(iree_hal_device_t** device);
 
+#ifndef IREE_FILE_TOC
+#define IREE_FILE_TOC
+typedef struct iree_file_toc_t {
+  const char* name;  // the file's original name
+  const char* data;  // beginning of the file
+  size_t size;       // length of the file
+} iree_file_toc_t;
+#endif  // IREE_FILE_TOC
+
+// A function to load the vm bytecode module from the different backend targets.
+// The bytecode module is generated for the specific backend and platform.
+extern const struct iree_file_toc_t* load_bytecode_module();
```

Comment:
we should change this to return an `iree_const_byte_span_t` so we don't have the file toc leak across the boundary

Reply:
Great suggestion! Done.